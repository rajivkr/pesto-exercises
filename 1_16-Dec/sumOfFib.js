let fibo = n => {
  if (n <= 1) return n;
  return fibo(n - 1) + fibo(n - 2);
};

let sumOfFibs = num => {
  let result = 0,
    i = 1;
  while (result < num) {
    let currentFib = fibo(i);
    if (currentFib % 2) {
      result += currentFib;
    }
    i++;
  }
  return result;
};

export { sumOfFibs };
