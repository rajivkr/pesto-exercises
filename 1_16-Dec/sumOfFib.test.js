import { sumOfFibs } from './sumOfFib';

describe('sum of odd Fib test cases', () => {
  it('should add Odd fib until given number', () => {
    expect(sumOfFibs(20)).toMatchSnapshot();
    expect(sumOfFibs(30)).toMatchSnapshot();
  });
});
