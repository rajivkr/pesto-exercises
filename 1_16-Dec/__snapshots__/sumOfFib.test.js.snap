// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sum of odd Fib test cases should add Odd fib until given number 1`] = `23`;

exports[`sum of odd Fib test cases should add Odd fib until given number 2`] = `44`;
